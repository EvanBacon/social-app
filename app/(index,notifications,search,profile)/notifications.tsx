export {NotificationsScreen as default} from 'view/screens/Notifications'
