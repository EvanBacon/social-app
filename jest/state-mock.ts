import {RootStoreModel} from './../src/state/models/root-store'
import {NavigationTabModel} from './../src/state/models/navigation'
import {SessionModel} from '../src/state/models/session'
import {NavigationModel} from '../src/state/models/navigation'
import {ShellUiModel} from '../src/state/models/shell-ui'
import {MeModel} from '../src/state/models/me'
import {OnboardModel} from '../src/state/models/onboard'
import {ProfilesViewModel} from '../src/state/models/profiles-view'
import {LinkMetasViewModel} from '../src/state/models/link-metas-view'
import {MembershipsViewModel} from '../src/state/models/memberships-view'
import {FeedModel} from '../src/state/models/feed-view'
import {NotificationsViewModel} from '../src/state/models/notifications-view'
import {LRUMap} from 'lru_map'

const mockedSessionStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  data: {
    service: '',
    refreshJwt: '',
    accessJwt: '',
    handle: '',
    did: '',
  },
  online: false,
  attemptingConnect: false,
  rootStore: {} as RootStoreModel,
  hasSession: true,
  clear: jest.fn(),
  setState: jest.fn(),
  setOnline: jest.fn(),
  updateAuthTokens: jest.fn(),
  connect: jest.fn(),
  describeService: jest.fn(),
  login: jest.fn(),
  createAccount: jest.fn(),
  logout: jest.fn(),

  // unknown added because of the missing private methods: _connectPromise, configureApi & _connect
} as unknown as SessionModel

const mockedNavigationTabStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  id: 0,
  history: [
    {
      url: '',
      ts: 0,
      title: '',
      id: 0,
    },
  ],
  index: 0,
  isNewTab: false,
  current: {
    url: '',
    ts: 0,
    title: '',
    id: 0,
  },
  canGoBack: false,
  canGoForward: false,
  backTen: [
    {
      url: '',
      title: '',
      index: 0,
      id: 0,
    },
  ],
  forwardTen: [
    {
      url: '',
      title: '',
      index: 0,
      id: 0,
    },
  ],
  navigate: jest.fn(),
  refresh: jest.fn(),
  goBack: jest.fn(),
  fixedTabReset: jest.fn(),
  goForward: jest.fn(),
  goToIndex: jest.fn(),
  setTitle: jest.fn(),
  setIsNewTab: jest.fn(),
  fixedTabPurpose: 0,
  getBackList: jest.fn(),
  getForwardList: jest.fn(),
} as NavigationTabModel

const mockedNavigationStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  tabs: [],
  tabIndex: 0,
  clear: jest.fn(),
  tab: mockedNavigationTabStore,
  tabCount: 1,
  isCurrentScreen: jest.fn(),
  navigate: jest.fn(),
  refresh: jest.fn(),
  setTitle: jest.fn(),
  handleLink: jest.fn(),
  switchTo: jest.fn(),
  setActiveTab: jest.fn(),
  closeTab: jest.fn(),
  newTab: jest.fn(),
} as NavigationModel

const mockedShellStore = {
  minimalShellMode: false,
  isMainMenuOpen: false,
  isModalActive: false,
  activeModal: undefined,
  isLightboxActive: false,
  activeLightbox: undefined,
  isComposerActive: false,
  composerOpts: undefined,
  setMainMenuOpen: jest.fn(),
  setMinimalShellMode: jest.fn(),
  openModal: jest.fn(),
  closeModal: jest.fn(),
  closeComposer: jest.fn(),
  closeLightbox: jest.fn(),
  openComposer: jest.fn(),
  openLightbox: jest.fn(),
} as ShellUiModel

const mockedMeStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  did: '',
  handle: '',
  displayName: '',
  description: '',
  avatar: '',
  notificationCount: 0,
  rootStore: {} as RootStoreModel,
  // TODO: mock these models
  memberships: {
    placeholder: null,
  } as MembershipsViewModel,
  mainFeed: {
    placeholder: null,
  } as FeedModel,
  notifications: {
    placeholder: null,
  } as NotificationsViewModel,
  clear: jest.fn(),
  load: jest.fn(),
  clearNotificationCount: jest.fn(),
  fetchStateUpdate: jest.fn(),
  refreshMemberships: jest.fn(),
} as MeModel

const mockedOnboardStore = {
  serialize: jest.fn(),
  hydrate: jest.fn(),
  isOnboarding: false,
  stage: '',
  start: jest.fn(),
  stop: jest.fn(),
  next: jest.fn(),
} as OnboardModel

const mockedProfileStore = {
  hydrate: jest.fn(),
  serialize: jest.fn(),
  cache: new LRUMap(100),
  rootStore: {} as RootStoreModel,
  getProfile: jest.fn(),
  overwrite: jest.fn(),
} as ProfilesViewModel

const mockedLinkMetasStore = {
  hydrate: jest.fn(),
  serialize: jest.fn(),
  cache: new LRUMap(100),
  rootStore: {} as RootStoreModel,
  getLinkMeta: jest.fn(),
} as LinkMetasViewModel

export const mockedRootStore = {
  api: false,
  resolveName: jest.fn(),
  serialize: jest.fn(),
  hydrate: jest.fn(),
  fetchStateUpdate: jest.fn(),
  clearAll: jest.fn(),
  session: mockedSessionStore,
  nav: mockedNavigationStore,
  shell: mockedShellStore,
  me: mockedMeStore,
  onboard: mockedOnboardStore,
  profiles: mockedProfileStore,
  linkMetas: mockedLinkMetasStore,
}
