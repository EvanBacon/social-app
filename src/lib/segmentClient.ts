import {createClient} from '@segment/analytics-react-native'
// import {createLogger} from './logger'

export const segmentClient = createClient({
  writeKey: '8I6DsgfiSLuoONyaunGoiQM7A6y2ybdI',
  trackAppLifecycleEvents: true,
  // Uncomment to debug:
  // logger: createLogger(),
})
