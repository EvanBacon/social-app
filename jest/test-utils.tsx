import React from 'react'
import {render} from '@testing-library/react-native'
import {GestureHandlerRootView} from 'react-native-gesture-handler'
import {RootSiblingParent} from 'react-native-root-siblings'
import {SafeAreaProvider} from 'react-native-safe-area-context'
import {RootStoreModel, RootStoreProvider} from '../src/state'
import {mockedRootStore} from './state-mock'

const WrappedComponent = ({children}: any) => {
  return (
    // eslint-disable-next-line react-native/no-inline-styles
    <GestureHandlerRootView style={{flex: 1}}>
      <RootSiblingParent>
        <RootStoreProvider value={mockedRootStore as RootStoreModel}>
          <SafeAreaProvider>{children}</SafeAreaProvider>
        </RootStoreProvider>
      </RootSiblingParent>
    </GestureHandlerRootView>
  )
}

const customRender = (ui: any, options?: any) =>
  render(ui, {wrapper: WrappedComponent, ...options})

// re-export everything
export * from '@testing-library/react-native'

// override render method
export {customRender as render}
