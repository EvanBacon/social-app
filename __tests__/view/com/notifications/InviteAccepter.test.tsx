import React from 'react'
import {fireEvent, render, waitFor} from '../../../../jest/test-utils'
import {ConfirmModal} from '../../../../src/state/models/shell-ui'
import {InviteAccepter} from '../../../../src/view/com/notifications/InviteAccepter'
import {
  mockedMembershipsModel,
  mockedMeStore,
  mockedNotificationsViewItemModel,
  mockedRootStore,
  mockedShellStore,
} from '../../../../__mocks__/state-mock'
import * as apilib from '../../../../src/state/lib/api'
import * as Toast from '../../../../src/view/com/util/Toast'

describe('InviteAccepter', () => {
  const mockedProps = {
    item: mockedNotificationsViewItemModel,
  }
  afterAll(() => {
    jest.clearAllMocks()
  })

  it('presses accept invite button', async () => {
    const openModalMock = jest
      .fn()
      .mockImplementation((item: ConfirmModal) => item.onPressConfirm())
    const spyOnToast = jest.spyOn(Toast, 'show')
    const spyOnAcceptSceneInvite = jest
      .spyOn(apilib, 'acceptSceneInvite')
      .mockResolvedValue('test ok')
    const {findByTestId} = render(<InviteAccepter {...mockedProps} />, {
      ...mockedRootStore,
      shell: {
        ...mockedShellStore,
        openModal: openModalMock,
      },
    })

    const acceptInviteButton = await findByTestId('acceptInviteButton')
    expect(acceptInviteButton).toBeTruthy()
    fireEvent.press(acceptInviteButton)

    expect(openModalMock).toHaveBeenCalled()
    expect(spyOnAcceptSceneInvite).toHaveBeenCalled()

    await waitFor(() => {
      expect(mockedMeStore.refreshMemberships).toHaveBeenCalled()
      expect(spyOnToast).toHaveBeenCalled()
    })
  })

  it('renders invite accepted message when already member', async () => {
    jest.spyOn(React, 'useState').mockReturnValue(['123', jest.fn()])
    const {findByTestId} = render(<InviteAccepter {...mockedProps} />, {
      ...mockedRootStore,
      me: {
        ...mockedMeStore,
        memberships: {
          ...mockedMembershipsModel,
          isMemberOf: jest.fn().mockReturnValue(true),
        },
      },
    })
    const inviteAccepted = await findByTestId('inviteAccepted')
    expect(inviteAccepted).toBeTruthy()
  })

  it('matches snapshot', () => {
    const page = render(<InviteAccepter {...mockedProps} />)
    expect(page).toMatchSnapshot()
  })
})
