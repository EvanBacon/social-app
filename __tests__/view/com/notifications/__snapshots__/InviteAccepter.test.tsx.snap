// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InviteAccepter matches snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
      testID="inviteAccepted"
    >
      <
        icon="check"
        size={14}
        style={
          Object {
            "marginRight": 5,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
            },
            Object {
              "fontSize": 16,
              "letterSpacing": 0.25,
            },
            Array [
              Object {
                "color": "#645454",
              },
              Object {
                "fontSize": 15,
              },
            ],
          ]
        }
      >
        Invite accepted
      </Text>
    </View>
  </View>
</View>
`;
